txn.go

```diff
@@ -184,12 +177,50 @@ func (o *oracle) newCommitTs(txn *Txn) uint64 {
 		ts = txn.commitTs
 	}
 
-	for _, w := range txn.writes {
-		o.commits[w] = ts // Update the commitTs.
+	if ts > o.lastCleanupTs {
```

Comment:
You are right that this is suspicious-looking. I think I was under the (mistaken) assumption that there could be case where the commit timestamp would not increase, but obviously that would break assumptions all over the place.

Reply:
Yes. If the new timestamp is smaller than the previous one, it would mess up the look-ups. We assume that newer values with be at a higher level (level 0, level 1, etc) with higher timestamps.

Let's just complain to the user in that case. This could potentially mean there's something seriously wrong with badger or whoever is using badger. We shouldn't quitely continue here.